python/adjoint/filter_source.py

```diff
@@ -103,6 +109,21 @@ def nuttall_dtft(self, f, f0):
         a = [0.355768, 0.4873960, 0.144232, 0.012604]
         return self.cos_window_fd(a, f, f0)
 
+    ## compute the bandwidth of the DTFT of the Nuttall window function
+    ## (magnitude) assuming it has decayed from its peak value by some
+    ## tolerance by fitting it to an asymptotic power law of the form
+    ## C * (Δf / f)^3 where C is a constant, Δf is the frequency bandwidth
+    ## of the Nuttall window, and f is the frequency
+    def nuttall_bandwidth(self):
+        tol = 1e-7
+        fwidth = 1/(self.N * self.dt)
+        frq_inf = 1000*fwidth
+        na_dtft = self.nuttall_dtft(frq_inf, 0)
+        coeff = frq_inf**3 * np.abs(na_dtft) * 1/fwidth**3
```

Comment:
> Note that coefficient C in `C * (Δf/f)^3` should not change when you change Δf (i.e. N or dt); otherwise, you have the wrong scaling.

In the way it is currently setup on this line, the coefficient C (`coeff`) *does* depend on Δf (`fwidth`): C  = (f / Δf)^3 |W(f)|. f in the asymptotic limit (`frq_inf`) is chosen arbitrarily as 1000Δf.

Reply:
My thinking was that you should be able to compute it once and for all, get the coefficient (appropriately scaled by the fwidth), and hard-code it in the source — that way you don't need to compute `nuttall_dtft` at `frq_inf` in the source.